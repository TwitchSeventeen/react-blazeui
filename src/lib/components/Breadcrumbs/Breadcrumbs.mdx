---
name: Breadcrumbs
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Breadcrumbs from './index';

# Breadcrumbs

Breadcrumbs are useful to indicate the navigational route into a website.

## Basic usage
Use the **active** prop to style the active item. Every other item will get a divider placed after its child.
<Playground>
    <Breadcrumbs>
        <Breadcrumbs.Item>
            <a class="c-link" href="#">lorem</a>
        </Breadcrumbs.Item>
        <Breadcrumbs.Item>
            <a class="c-link" href="#">ipsum</a>
        </Breadcrumbs.Item>
        <Breadcrumbs.Item active>
            dolor
        </Breadcrumbs.Item>
    </Breadcrumbs>
</Playground>

## Properties
### Breadcrumbs
<PropsTable of={Breadcrumbs} />

### Breadcrumbs.Item
<PropsTable of={Breadcrumbs.Item} />