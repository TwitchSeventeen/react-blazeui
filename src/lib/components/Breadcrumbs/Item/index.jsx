import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Item(props) {
  const {
    children, active, className, dataName, ...other
  } = props;
  const classes = classNames('c-breadcrumbs__crumb', className);
  return (
    <li
      {...other}
      className={classes}
      aria-current={active ? 'page' : null}
      data-name={dataName}
    >
      { children }
    </li>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Whether or not this item is active
   */
  active: PropTypes.bool,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Item.defaultProps = {
  children: null,
  active: false,
  className: null,
  dataName: 'breadcrumbs-item',
};

export default Item;
