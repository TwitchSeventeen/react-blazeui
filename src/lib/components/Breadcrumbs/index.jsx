import React from 'react';
import PropTypes from 'prop-types';
import Item from './Item';

function Breadcrumbs(props) {
  const { children, dataName, ...other } = props;
  return (
    <nav
      {...other}
      aria-label="Breadcrumb"
      data-name={dataName}
    >
      <ol className="c-breadcrumbs">
        { children }
      </ol>
    </nav>
  );
}

Breadcrumbs.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Breadcrumbs.defaultProps = {
  children: null,
  dataName: 'breadcrumbs',
};

Breadcrumbs.Item = Item;

export default Breadcrumbs;
