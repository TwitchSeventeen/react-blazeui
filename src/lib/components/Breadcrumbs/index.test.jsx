import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Breadcrumbs from './index';

describe('Breadcrumbs', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Breadcrumbs).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Breadcrumbs>
        <Breadcrumbs.Item>
          <a href="/lorem">lorem</a>
        </Breadcrumbs.Item>
        <Breadcrumbs.Item>
          <a href="/ipsum">ipsum</a>
        </Breadcrumbs.Item>
        <Breadcrumbs.Item active>
          dolor
        </Breadcrumbs.Item>
      </Breadcrumbs>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
