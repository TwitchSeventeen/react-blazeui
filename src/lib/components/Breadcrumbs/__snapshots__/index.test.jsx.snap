// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders correctly 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    data-name="breadcrumbs"
  >
    <ol
      class="c-breadcrumbs"
    >
      <li
        class="c-breadcrumbs__crumb"
        data-name="breadcrumbs-item"
      >
        <a
          href="/lorem"
        >
          lorem
        </a>
      </li>
      <li
        class="c-breadcrumbs__crumb"
        data-name="breadcrumbs-item"
      >
        <a
          href="/ipsum"
        >
          ipsum
        </a>
      </li>
      <li
        aria-current="page"
        class="c-breadcrumbs__crumb"
        data-name="breadcrumbs-item"
      >
        dolor
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
